pos_accounting/src/pool/storage/operator_impls.rs

```diff
@@ -301,7 +310,10 @@ impl<S: PoSAccountingStorageWrite<T>, T: StorageTag> PoSAccountingDB<S, T> {
 
         self.add_to_delegation_balance(undo_data.delegation_id, undo_data.amount)?;
 
-        self.add_balance_to_pool(pool_id, undo_data.amount)?;
+        // it's possible that the pool was decommissioned
+        if self.pool_exists(pool_id)? {
+            self.add_balance_to_pool(pool_id, undo_data.amount)?;
+        }
```

Comment:
Is it legal to have a non-existing pool here? I.e. should there be an `else` with an error?
This is replicated in multiple places, could the code somehow be shared?

Reply:
It is legal. The reason is that decommissioning pool removes only pool balance and data. Pool's owner cannot affect delegations, so they are kept in the db until delegation owner decides to spend it.